grpc-protocol/src/main/java/com/linecorp/armeria/common/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -91,21 +94,31 @@ public UnaryGrpcClient(HttpClient httpClient) {
                                          "Non-successful HTTP response code: " + msg.status());
                              }
 
-                             final String grpcStatus = msg.headers().get(GrpcHeaderNames.GRPC_STATUS);
-                             if (grpcStatus != null && !"0".equals(grpcStatus)) {
-                                 String grpcMessage = msg.headers().get(GrpcHeaderNames.GRPC_MESSAGE);
-                                 if (grpcMessage != null) {
-                                     grpcMessage = StatusMessageEscaper.unescape(grpcMessage);
-                                 }
-                                 throw new ArmeriaStatusException(
-                                         Integer.parseInt(grpcStatus),
-                                         grpcMessage);
+                             // Status can either be in the headers or trailers depending on error
+                             String grpcStatus = msg.headers().get(GrpcHeaderNames.GRPC_STATUS);
+                             if (grpcStatus != null) {
```

Comment:
Think a bit cleaner as

```java
if (grpcStatus == null) {
  grpcStatus = trailers.get(
```

Reply:
but we would still have to flip which we use to get the other header right? this also controls the condition of whether you get GRPC_MESSAGE from headers or trailers

Reply:
or are you saying having the same blocks flipped in order is the cleaner thing..

Reply:
Ah nevermind that's right. Let's extend the test to also have a grpc-message so I don't accidentally break it ;)

Reply:
sgtm